cypress/integration/frontsite/details.test.spec.js

```diff
@@ -0,0 +1,104 @@
+//https://127.0.0.1:3000/search?f=doctype%3ABOOK
+
+describe('frontsite literature details', () => {
+  const goToSearchDocument = () => {
+    // Books + available for loan
+    cy.visit(
+      '/search?f=doctype%3ABOOK&f=availability%3Aavailable%20for%20loan'
+    );
+  };
+
+  it('can view document details and request loan if available', () => {
+    cy.server();
+    cy.route('GET', '/api/locations/*').as('location');
+    cy.route('GET', '/api/me/loans?document_pid=*').as('loan');
+    cy.route('GET', '/api/document-requests/*').as('requests');
+
+    goToSearchDocument();
+
+    cy.get('.fs-book-card')
+      .first()
+      .click();
+
+    cy.url().should('contain', Cypress.config().baseUrl + '/literature/');
+
+    cy.wait('@loan').should(xhr => expect(xhr.status).to.eq(401));
+
+    cy.contains('Sign in to loan').click();
+
+    cy.get('input#email').type('patron1@test.ch');
+    cy.get('input#password').type('123456');
+    cy.contains('button.primary', 'Sign in').click();
+
+    cy.url().should('contain', Cypress.config().baseUrl + '/literature/');
+
+    cy.contains('Yannic Vilma').click();
+    cy.contains('Your loans').click();
+    cy.wait('@requests');
+    cy.get(':nth-child(2).statistic > .value').then($stat1 => {
+      let expectedNextCount = parseInt($stat1.text());
+
+      cy.go('back');
+
+      cy.wait('@loan').then(xhr => {
+        const {
+          has_active_loan: hasActiveLoan,
+          is_requested: isRequested,
+        } = xhr.response.body;
+        expect(hasActiveLoan).to.be.false;
+        if (isRequested) {
+          cy.contains('Have it delivered to my office').click();
+
+          cy.get('input[name=selectedDate]').click();
+
+          cy.wait('@location').wait(1000); // Cannot avoid this delay
+          cy.get('td:not(.disabled) > span.suicr-content-item')
```

Comment:
I think it has to do with how `semantic-ui-react-calendar` is implemented because I had some inconsistencies when removing the delay (to be investigated further of course). Also a few lines above this, someone had a similar issue with another calendar library ([here](https://www.cypress.io/blog/2019/01/22/when-can-the-test-click/)), the click is registered, the calendar is dismissed but not date is selected; had to use a workaround.